src/utils/build-info.ts

```diff
@@ -122,3 +123,34 @@ command = "${chosenSettings.devCommand}"
     return chosenSettings
   }
 }
+
+/**
+ * Merges the settings from the heuristics with the cached config.
+ * Returns the updated cached config
+ * @param cachedConfig The cached config
+ * @param settings The settings from the heuristics
+ */
+export const mergeSettingsWithCachedConfig = (cachedConfig: $TSFixMe, settings?: Settings): $TSFixMe => {
+  if (!settings) {
+    return cachedConfig
+  }
+
+  // override the build command with the detection result if no command is specified through the config
+  if (!cachedConfig.config.build.command) {
+    cachedConfig.config.build.command = settings.buildCommand
+    cachedConfig.config.build.commandOrigin = 'heuristics'
+  }
+
+  if (cachedConfig.config.build.publishOrigin === 'default') {
+    cachedConfig.config.build.publish = settings.dist
+    cachedConfig.config.build.publishOrigin = 'heuristics'
+  }
+
+  settings.plugins_recommended.forEach((plugin) => {
+    if (!cachedConfig.config.plugins.includes(plugin)) {
```

Comment:
How does this check work? We're pushing objects such as `{ package: "package-name" }` to `cachedConfig.config.plugins`, so how can we use `includes` to check if a plugin is already there?

Reply:
that happens if you don't have types and everything is any 😬